apps/condo/domains/ticket/utils/handlers.js

```diff
@@ -113,12 +136,86 @@ const handleTicketEvents = async (requestData) => {
     }
 
     if (eventTypes[STATUS_CHANGED_EVENT_TYPE]) {
-        // TODO(DOMA-2434): Add logic for sending notifications to resident on ticket status change
+        let type
+        switch (nextStatusId) {
+            case STATUS_IDS.OPEN:
+                if (statusReopenedCounter > 0)
+                    type = updatedBy !== client && TICKET_STATUS_RETURNED_TYPE
+                else
+                    type = createdBy !== client && TICKET_STATUS_OPENED_TYPE
+                break
+
+            case STATUS_IDS.IN_PROGRESS:
+                type = TICKET_STATUS_IN_PROGRESS_TYPE
+                break
+
+            case STATUS_IDS.COMPLETED:
+                type = TICKET_STATUS_COMPLETED_TYPE
+                break
+
+            case STATUS_IDS.DECLINED:
+                type = TICKET_STATUS_DECLINED_TYPE
+                break
+        }
+
+        if (type) {
+            await sendMessage(context, {
+                lang,
+                to: { user: { id: client } },
+                type,
+                meta: {
+                    dv: 1,
+                    data: {
+                        ticketId: updatedItem.id,
+                        ticketNumber: updatedItem.number,
+                        userId: client,
+                        domain: TICKET_DOMAIN,
+                    },
+                },
+                sender: updatedItem.sender,
+            })
+        }
+    }
+}
+
+const handleTicketCommentEvents = async (requestData) => {
+    const { updatedItem, context } = requestData
+    const createdBy = get(updatedItem, 'createdBy')
+
+    const [ticket] = await Ticket.getAll(context, { id: updatedItem.ticket })
+    const client = get(ticket, 'client.id')
+    const organizationId = get(ticket, 'organization.id')
+
+    const organization = await getByCondition('Organization', {
+        id: organizationId,
+        deletedAt: null,
+    })
+
+    const lang = get(COUNTRIES, [organization.country, 'locale'], DEFAULT_LOCALE)
+
+    if (client && createdBy !== client) {
+        await sendMessage(context, {
+            lang,
+            to: { user: { id: client } },
+            type: TICKET_COMMENT_ADDED_TYPE,
+            meta: {
+                dv: 1,
+                data: {
+                    ticketId: ticket.id,
+                    ticketNumber: ticket.number,
+                    userId: client,
+                    commentId: updatedItem.id,
+                    domain: TICKET_COMMENT_DOMAIN,
+                },
+            },
+            sender: updatedItem.sender,
```

Comment:
!! requestData.sender

Reply:
If I understand correctly, there is no sender field in request Data

Reply:
You mast require this field!